crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -3,14 +3,114 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use syn::{parse::ParseStream, Attribute, Ident, Result, Token, Type};
+use std::collections::HashSet;
+
+use syn::{
+    parse::{Error, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    Attribute, Expr, Ident, Meta, MetaNameValue, Path, Result, Token, Type,
+};
+
+use quote::format_ident;
+
+/// An optional identifier of the functions passed with flags
+/// e.g. read = my_getter, IdentFlag would be used to store my_getter
+type IdentFlag = Option<Ident>;
+
+/// Struct for storing the flags provided for a QProperty and their optional identifiers ([IdentFlag])
+#[derive(Debug)]
+pub struct QPropertyFlags {
+    read: IdentFlag, // TODO: maybe change this to better represent the data
+    write: Option<IdentFlag>,
+    notify: Option<IdentFlag>,
+}
+
+impl QPropertyFlags {
+    pub fn new() -> Self {
+        Self {
+            read: None,
+            write: None,
+            notify: None,
+        }
+    }
+
+    pub fn all_flags() -> Self {
+        Self {
+            read: None,
+            write: Some(None),
+            notify: Some(None),
+        }
+    }
+
+    fn set_field_by_string(&mut self, key: String, value: IdentFlag) -> Result<String> {
+        match key.as_str() {
+            "read" => self.read = value,
+            "write" => self.write = Some(value),
+            "notify" => self.notify = Some(value),
+            _ => {
+                return Err(Error::new(
+                    format_ident!("{}", key).span(), // TODO: check if this is an acceptable form of erroring for non Syn functions
```

Comment:
i don't think this will point to the original span as the information has been lost, instead consider passing in the `Ident` type as the `key` not the `String` then you can use `key.span()`

Reply:
done